File: py/42.py
from math import sqrt
from re import findall

def value(word):
    return reduce(lambda a, b: a + b,
             map(lambda c: ord(c) - 64, list(word)))

def triangle(n):
    return sqrt(n * 8 + 1) % 1 == 0

with open('houston/input/42') as f:
    words = sorted(findall('\w+', f.read()))
    print len(filter(triangle, map(value, words)))

File: js/42.js
var fs = require('fs'),
    input = fs.readFileSync('../input/42'),
    words = eval('[' + input + ']');

function triangular(n) {
    return Math.sqrt(8 * n + 1) % 1 == 0;
}

function value(word) {
    return word.split('').map(function(c) {
        return c.charCodeAt(0) - 64;
    }).reduce(function(a, b) {
        return a + b;
    });
}

console.log(words.map(value).filter(triangular).length);
